clang/test/SemaSYCL/deferred-diagnostics-aux-builtin.cpp

```diff
@@ -0,0 +1,47 @@
+// RUN: %clang_cc1 -triple spir64-unknown-unknown-sycldevice -fsycl -fsycl-is-device -aux-triple x86_64-unknown-linux-gnu -verify -fsyntax-only  %s
+//
+/*
+ This test is to verify that deferred diagnostics are emitted whenever there is an AUX target builtin function inside device code.
+ x86_64 is the AUX target. Spir64 is the device target.
+ _mm_prefetch is the AUX target builtin.
+*/
+
+// Testing that the deferred diagnostics work in conjunction with the SYCL namespaces.
+inline namespace cl {
+namespace sycl {
+
+template <typename name, typename Func>
+__attribute__((sycl_kernel)) void kernel_single_task(Func kernelFunc) {
+  // expected-note@+1 {{called by 'kernel_single_task<AName, (lambda}}
+  kernelFunc();
+}
+
+} // namespace sycl
+} // namespace cl
+
+int main(int argc, char **argv) {
+
+  //This is host code. This will not be compiled for the device.
+  /* 
+  _mm_prefetch is an x86-64 target architecture built-in function.
```

Comment:
The rest of this comment is unnecessary.  Just something like:

_mm_prefetch is an x86-64 specific builtin where the second integer parameter is required to be a constant between 0 and 7.  We still need to ensure we validate errors in device mode, as well as diagnose uses in kernel code.